osu.Game/Graphics/Containers/BeatSyncedContainer.cs

```diff
@@ -128,20 +127,22 @@ protected override void Update()
 
             TimeSinceLastBeat = beatLength - TimeUntilNextBeat;
 
-            if (ReferenceEquals(timingPoint, LastTimingPoint) && beatIndex == lastBeat)
+            if (ReferenceEquals(timingPoint, lastTimingPoint) && beatIndex == lastBeat)
                 return;
 
             // as this event is sometimes used for sound triggers where `BeginDelayedSequence` has no effect, avoid firing it if too far away from the beat.
             // this can happen after a seek operation.
             if (AllowMistimedEventFiring || Math.Abs(TimeSinceLastBeat) < MISTIMED_ALLOWANCE)
             {
                 using (BeginDelayedSequence(-TimeSinceLastBeat))
-                    OnNewBeat(beatIndex, timingPoint, effectPoint, BeatSyncSource.Amplitudes ?? ChannelAmplitudes.Empty);
+                    OnNewBeat(beatIndex, timingPoint, effectPoint, BeatSyncSource.CurrentAmplitudes);
             }
 
             lastBeat = beatIndex;
-            LastTimingPoint = timingPoint;
-            LastEffectPoint = effectPoint;
+            lastTimingPoint = timingPoint;
+            lastEffectPoint = effectPoint;
+
+            IsKiaiTime = lastEffectPoint?.KiaiMode ?? false;
```

Comment:
nit: Would find it more sensible if `IsKiaiTime` used the current effect point rather than "last" after it has been updated to current, i.e.:
```csharp
IsKiaiTime = effectPoint?.KiaiMode ?? false;

lastBeat = beatIndex;
lastTimingPoint = timingPoint;
lastEffectPoint = effectPoint;
```

Reply:
🤷